packages/plugin-legacy/src/index.ts

```diff
@@ -648,7 +648,9 @@ async function buildPolyfillChunk(
   excludeSystemJS?: boolean
 ) {
   let { minify, assetsDir } = buildOptions
-  minify = minify ? 'terser' : false
+  if (format === 'iife') {
+    minify = minify ? 'terser' : false
+  }
```

Comment:
Note: This function is only called in two places. One for building the modern polyfilll chunk, two for legacy polyfill chunk. They pass `es` and `iife` respectively. So this changes that modern polyfills uses esbuild instead, which should be fine.